lib/SILOptimizer/Differentiation/PullbackCloner.cpp

```diff
@@ -400,12 +458,45 @@ class PullbackCloner::Implementation final
     }
     /// If adjoint value is concrete, it is already materialized. Store it in
     /// the destination address.
-    case AdjointValueKind::Concrete:
+    case AdjointValueKind::Concrete: {
       auto concreteVal = val.getConcreteValue();
       builder.emitStoreValueOperation(loc, concreteVal, destAddress,
                                       StoreOwnershipQualifier::Init);
       break;
     }
+    case AdjointValueKind::AddElement: {
+      auto *addElementValue = val.getAddElementValue();
+      auto baseAdjoint = addElementValue->baseAdjoint;
+      auto eltToAdd = addElementValue->eltToAdd;
+
+      builder.emitZeroIntoBuffer(loc, destAddress, IsInitialization);
```

Comment:
Shouldn't you do indirect materialization of `baseAdjoint` into `destAddress?

Reply:
You're right. Changed it.